src/reference/antora/modules/ROOT/pages/jdbc/message-store-json.adoc

```diff
@@ -0,0 +1,257 @@
+      PRIMARY KEY (REGION, GROUP_KEY, CREATED_DATE, MESSAGE_SEQUENCE)
+);
+----
+
+=== MySQL
+
+[source,sql]
+----
+CREATE TABLE JSON_CHANNEL_MESSAGE (
+   MESSAGE_ID CHAR(36) NOT NULL,
+   GROUP_KEY CHAR(36) NOT NULL,
+   CREATED_DATE BIGINT NOT NULL,
+   MESSAGE_PRIORITY BIGINT,
+   MESSAGE_SEQUENCE BIGINT NOT NULL,
+   MESSAGE_CONTENT JSON, -- JSON message content
+   REGION VARCHAR(100) NOT NULL,
+   CONSTRAINT JSON_CHANNEL_MESSAGE_PK
+      PRIMARY KEY (REGION, GROUP_KEY, CREATED_DATE, MESSAGE_SEQUENCE)
+);
+----
+
+=== H2
+
+[source,sql]
+----
+CREATE TABLE JSON_CHANNEL_MESSAGE (
+   MESSAGE_ID CHAR(36) NOT NULL,
+   GROUP_KEY CHAR(36) NOT NULL,
+   CREATED_DATE BIGINT NOT NULL,
+   MESSAGE_PRIORITY BIGINT,
+   MESSAGE_SEQUENCE BIGINT NOT NULL,
+   MESSAGE_CONTENT CLOB, -- JSON message content
+   REGION VARCHAR(100) NOT NULL,
+   CONSTRAINT JSON_CHANNEL_MESSAGE_PK
+      PRIMARY KEY (REGION, GROUP_KEY, CREATED_DATE, MESSAGE_SEQUENCE)
+);
+----
+
+== JSON Structure
+
+Messages are stored with the following JSON structure:
+
+[source,json]
+----
+{
+  "@class": "org.springframework.messaging.support.GenericMessage",
+  "payload": {
+    "@class": "com.example.OrderMessage",
+    "orderId": "ORDER-12345",
+    "amount": 1299.99
+  },
+  "headers": {
+    "@class": "java.util.HashMap",
+    "priority": ["java.lang.String", "HIGH"],
+    "id": ["java.util.UUID", "a1b2c3d4-..."],
+    "timestamp": ["java.lang.Long", 1234567890]
+  }
+}
+----
+
+The `@class` properties provide type information necessary for proper deserialization
+of polymorphic types.
+
+== Querying JSON Content (Optional)
+
+If you use native JSON column types (PostgreSQL JSONB or MySQL JSON), you can
+query message content directly:
+
+=== PostgreSQL JSONB Queries
+
+[source,sql]
+----
+-- Find messages by payload field
+SELECT * FROM JSON_CHANNEL_MESSAGE
+WHERE MESSAGE_CONTENT @> '{"payload": {"orderId": "ORDER-12345"}}';
+
+-- Find high-priority messages
+SELECT * FROM JSON_CHANNEL_MESSAGE
+WHERE MESSAGE_CONTENT -> 'headers' @> '{"priority": ["java.lang.String", "HIGH"]}';
+----
+
+=== MySQL JSON Functions
+
+[source,sql]
+----
+-- Find messages by payload field
+SELECT * FROM JSON_CHANNEL_MESSAGE
+WHERE JSON_EXTRACT(MESSAGE_CONTENT, '$.payload.orderId') = 'ORDER-12345';
+
+-- Find high-priority messages
+SELECT * FROM JSON_CHANNEL_MESSAGE
+WHERE JSON_EXTRACT(MESSAGE_CONTENT, '$.headers.priority[1]') = 'HIGH';
+----
+
+[NOTE]
+====
+If you use `TEXT` or `CLOB` column types, these JSON-specific queries are not available,
+but the JSON serialization still works for storage and retrieval through Spring Integration.
+====
+
+== Trusted Packages
+
+The `JacksonMessagingUtils.messagingAwareMapper()` validates all deserialized classes
+against a trusted package list to prevent security vulnerabilities.
+
+Default trusted packages include:
+- `java.util`
+- `java.lang`
+- `org.springframework.messaging.support`
+- `org.springframework.integration.support`
```

Comment:
Doesn't look like the full list.
Please, copy it from the `JacksonMessagingUtils.DEFAULT_TRUSTED_PACKAGES`